### DISCUSSION TITLE
How to enforce document uniqueness?

### DISCUSSION THREAD

**Discussion by klacus** [ravendb/ravendb#15845] [Q&A] [answered]
In my project I need to enforce document uniqueness by a given field (email) in a document within a specific collection (accounts).
I tried to create the index in the RevenDB Studio, but could not figure out how to do it.
RavenDB version: 5.4.101

Do I need to change some configuration or I have a wrong understanding of how document uniqueness can be achieved in RavenDB?

**Comment by garayx** [comment, marked as answer]
Hi,

You can use a Compare Exchange to save a unique value in RavenDB, please see the following pages:
1. https://ravendb.net/docs/article-page/5.1/csharp/client-api/operations/compare-exchange/overview#compare-exchange-overview
2. https://ravendb.net/learn/inside-ravendb-book/reader/4.0/2-zero-to-ravendb

Another option is using cluster-wide transaction:
1. https://ravendb.net/docs/article-page/5.1/csharp/server/clustering/cluster-transactions#cluster-cluster-wide-transactions
2. https://ayende.com/blog/194405-A/ravendb-5-2-simplifying-atomic-cluster-wide-transactions
